src/server/connection.py

```diff
@@ -144,7 +165,8 @@ def add(self, connection: ChiaConnection) -> bool:
 
         if connection.connection_type == NodeType.FULL_NODE:
             self._state_changed("add_connection")
-            return self.peers.add(connection.get_peer_info())
+            if self.introducer_peers is not None:
+                return self.introducer_peers.add(connection.get_peer_info())
```

Comment:
Can `introducer_peers` just hook into the already existing `_state_changed` infrastructure rather than being hard-coded here?

Reply:
Introducer will go away in the next update (with DNS) right? So i don't think it's a big deal at this point. All the introducer code will be removed